corehq/apps/locations/models.py

```diff
@@ -295,15 +295,18 @@ def get_from_user_input(self, domain, user_input):
         except self.model.DoesNotExist:
             return self.get(domain=domain, name__iexact=user_input)
 
-    def filter_by_user_input(self, domain, user_input):
+    def filter_by_user_input(self, domain, user_input, from_location_ids=None):
         """
         Accepts partial matches, matches against name and site_code.
         """
-        return (self.filter(domain=domain)
-                    .filter(models.Q(name__icontains=user_input) |
-                            models.Q(site_code__icontains=user_input)))
-
-    def filter_path_by_user_input(self, domain, user_input):
+        locations = (self.filter(domain=domain)
+                     .filter(models.Q(name__icontains=user_input) |
+                             models.Q(site_code__icontains=user_input)))
+        if from_location_ids is not None:
+            locations = locations.filter(location_id__in=from_location_ids)
```

Comment:
I'd prefer just doing this as a separate filter - it feels like a second thing.  Instead of:
```python
SQLLocation.active_objects.filter_by_user_input(domain, user_input, from_location_ids)
```
You could just do:
```python
(SQLLocation.active_objects.filter_by_user_input(domain, user_input)
                           .filter(location_id__in=from_location_ids))
```

Reply:
Yup! that looks better.

Reply:
checked this, if from_locations_ids is None/[] in case we don't want to pick from specific set of locations, this logic would return no locations. So looks like the check for `from_location_ids` is still needed.